File: src/main/java/me/kaotich00/fwauctionhouse/commands/user/ConfirmCommand.java
package me.kaotich00.fwauctionhouse.commands.user;

import me.kaotich00.fwauctionhouse.FwAuctionHouse;
import me.kaotich00.fwauctionhouse.commands.api.UserCommand;
import me.kaotich00.fwauctionhouse.message.Message;
import me.kaotich00.fwauctionhouse.objects.PendingSell;
import me.kaotich00.fwauctionhouse.services.SimpleMarketService;
import me.kaotich00.fwauctionhouse.storage.StorageFactory;
import org.bukkit.Sound;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

import java.util.Optional;
import java.util.concurrent.CompletableFuture;

public class ConfirmCommand extends UserCommand {

    @Override
    public void onCommand(CommandSender sender, String[] args) {
        super.onCommand(sender, args);

        Player buyer = (Player) sender;

        String listingId = args[1];

        Optional<PendingSell> optPendingSell = SimpleMarketService.getInstance().getPendingSell(Integer.parseInt(listingId));
        if(optPendingSell.isPresent()) {
            PendingSell pendingSell = optPendingSell.get();

            ItemStack boughtItem = pendingSell.getItemStack();
            buyer.getInventory().addItem(boughtItem);

            FwAuctionHouse.getEconomy().withdrawPlayer(buyer, pendingSell.getTotalCost());

            Message.BOUGHT_ITEM.send(buyer, boughtItem.getI18NDisplayName(), boughtItem.getAmount(), pendingSell.getTotalCost());
            buyer.playSound(buyer.getLocation(), Sound.ENTITY_PLAYER_LEVELUP, 1, 1);

            CompletableFuture.runAsync(() -> {
                SimpleMarketService.getInstance().removeFromPendingSells(pendingSell);
                StorageFactory.getInstance().getStorageMethod().deletePendingSell(pendingSell.getListingId());
            });
        }
    }

    @Override
    public String getInfo() {
        return super.getInfo();
    }

    @Override
    public String getUsage() {
        return "/market confirm <listingId>";
    }

    @Override
    public String getName() {
        return super.getName();
    }

    @Override
    public Integer getRequiredArgs() {
        return 1;
    }

}

File: src/main/java/me/kaotich00/fwauctionhouse/commands/user/ConfirmCommand.kt
package me.kaotich00.fwauctionhouse.commands.user

import com.google.inject.Inject
import me.kaotich00.fwauctionhouse.FwAuctionHouse
import me.kaotich00.fwauctionhouse.commands.api.UserCommand
import me.kaotich00.fwauctionhouse.message.Message
import me.kaotich00.fwauctionhouse.services.ListingsService
import me.kaotich00.fwauctionhouse.storage.ListingsDao
import me.kaotich00.fwauctionhouse.utils.launchAsync
import org.bukkit.Sound
import org.bukkit.entity.Player

class ConfirmCommand @Inject constructor(
    private val listingsService: ListingsService,
    private val listingsDao: ListingsDao
) : UserCommand(
    "confirm",
    "",
    1,
    "/market confirm <listingId>"
) {

    override fun doCommand(sender: Player, args: Array<String>) {
        val listingId = args[1].toIntOrNull() ?: run {
            sender.sendMessage("You must insert a valid id")
            return
        }

        val pendingSell = listingsService.getPendingSell(listingId) ?: return

        val boughtItem = pendingSell.itemStack

        sender.inventory.addItem(boughtItem)

        FwAuctionHouse.economy.withdrawPlayer(sender, pendingSell.totalCost)

        Message.BOUGHT_ITEM.send(
            sender,
            boughtItem.i18NDisplayName ?: "N/D",
            boughtItem.amount,
            pendingSell.totalCost
        )

        sender.playSound(sender.location, Sound.ENTITY_PLAYER_LEVELUP, 1f, 1f)

        launchAsync {
            listingsService.removeFromPendingSells(pendingSell)
            listingsDao.deletePendingSell(pendingSell.listingId)
        }
    }

}